lib/spack/spack/relocate.py

```diff
@@ -0,0 +1,317 @@
+        if command.returncode != 0:
+            tty.warn('failed reading rpath for %s.' % path_name)
+            return False
+        return output.rstrip('\n').split(':')
+    else:
+        tty.die('relocation not supported for this platform')
+    return
+
+
+def get_relative_rpaths(path_name, orig_dir, orig_rpaths):
+    rel_rpaths = []
+    for rpath in orig_rpaths:
+        if re.match(orig_dir, rpath):
+            rel = os.path.relpath(rpath, start=os.path.dirname(path_name))
+            rel_rpaths.append('$ORIGIN/%s' % rel)
+        else:
+            rel_rpaths.append(rpath)
+    return rel_rpaths
+
+
+def macho_get_paths(path_name):
+    otool = which('otool')
+    output = otool("-l", path_name, output=str, err=str)
+    if otool.returncode != 0:
+        tty.warn('failed reading rpath for %s.' % path_name)
+        return False
+    last_cmd = None
+    idpath = ''
+    rpaths = []
+    deps = []
+    for line in output.split('\n'):
+        match = re.search('( *[a-zA-Z]+ )(.*)', line)
+        if match:
+            lhs = match.group(1).lstrip().rstrip()
+            rhs = match.group(2)
+            match2 = re.search('(.*) \(.*\)', rhs)
+            if match2:
+                rhs = match2.group(1)
+            if lhs == 'cmd':
+                last_cmd = rhs
+            if lhs == 'path' and last_cmd == 'LC_RPATH':
+                rpaths.append(rhs)
+            if lhs == 'name' and last_cmd == 'LC_ID_DYLIB':
+                idpath = rhs
+            if lhs == 'name' and last_cmd == 'LC_LOAD_DYLIB':
+                deps.append(rhs)
+    return rpaths, deps, idpath
+
+
+def macho_make_paths_rel(path_name, old_dir, rpaths, deps, idpath):
+    id = None
+    nrpaths = []
+    ndeps = []
+    if idpath:
+        id = '@rpath/%s' % os.path.basename(idpath)
+    for rpath in rpaths:
+        if re.match(old_dir, rpath):
+            rel = os.path.relpath(rpath, start=os.path.dirname(path_name))
+            nrpaths.append('@loader_path/%s' % rel)
+        else:
+            nrpaths.append(rpath)
+    for dep in deps:
+        if re.match(old_dir, dep):
+            rel = os.path.relpath(dep, start=os.path.dirname(path_name))
+            ndeps.append('@loader_path/%s' % rel)
+        else:
+            ndeps.append(dep)
+    return nrpaths, ndeps, id
+
+
+def macho_replace_paths(old_dir, new_dir, rpaths, deps, idpath):
+    id = None
+    nrpaths = []
+    ndeps = []
+    if idpath:
+        id = idpath.replace(old_dir, new_dir)
+    for rpath in rpaths:
+        nrpath = rpath.replace(old_dir, new_dir)
+        nrpaths.append(nrpath)
+    for dep in deps:
+        ndep = dep.replace(old_dir, new_dir)
+        ndeps.append(ndep)
+    return nrpaths, ndeps, id
+
+
+def modify_macho_object(path_name, old_dir, new_dir, relative):
+    """
+    Modify MachO binaries by changing rpaths,and id and dependency lib paths.
+    Examines the output of otool -l for these three patterns
```

Comment:
Someone asked to see the output of otool -l to understand what I am parsing. I removed this.